hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -661,10 +667,10 @@ private class Emit(
             GetTupleElement(Ref("i-1", tt), tt.fields(0).index) -> GetTupleElement(Ref("i", tt), tt.fields(0).index)
         }
 
+        val compare2 = ApplyComparisonOp(EQWithNA(ktyp.virtualType), lastKey, currKey)
```

Comment:
missed one infer, this isn't right however, need an env with these fields bound. fixing now.

Reply:
Fixed. Interesting: there's nothing that actually triggers this code in our test suite

Reply:
Actually python tests hit this; interesting that IRSuite.tetGroupKeyBy doesn't; looked briefly; exec strategy covers it, maybe gets optimized out?

Reply:
added unoptimized compile testutil; did not apply to all jvmOnly, because a few tests fail (testScan, and one other). That is outside of the scope of this PR, but may be interesting.

Reply:
Pretty sure this is getting constant-folded away.

Adding a no-opt compile path is a good change.